rpc_util.go

```diff
@@ -307,40 +309,45 @@ func encode(c Codec, msg interface{}, cp Compressor, cbuf *bytes.Buffer, outPayl
 			outPayload.Data = b
 			outPayload.Length = len(b)
 		}
+
 		if cp != nil {
+			// pre-alloc place
+			cbuf.Write(make([]byte, 5))
```

Comment:
Be consistent about your use of constants -- use "payloadLen+sizeLen" where you have 5.